### PR TITLE
Fix Swift 2 CocoaPods support

### LINKED ISSUE #2250: Cocoa pods and Swift 2.0 - module not found
Hi there. Shortly:

0. Xcode 7 beta 3

1. Podfile:

```
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '8.0'
use_frameworks!
inhibit_all_warnings!

pod 'Realm', :git => 'https://github.com/realm/realm-cocoa.git', :branch => 'swift-2.0'
pod 'RealmSwift', :git => 'https://github.com/realm/realm-cocoa.git', :branch => 'swift-2.0'
```
2. pod install (with Cocoapods 0.38)

3. import RealmSwift anywhere in project's code

4. Result: 

zrzut ekranu 2015-07-20 o 21 06 58

This started happening after weekend. I can send you test project I have created to reproduce this issue.

very best!

### PR DESCRIPTION
**Pull request by segiddins**
This should _actually_ fix #2250, in addition to adding watchos support to the RealmSwift podspec

\c @jpsim

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -645,7 +645,9 @@ case "$COMMAND" in
     # CocoaPods
     ######################################
     "cocoapods-setup")
-        sh build.sh download-core
+        if [[ "$2" = "without-core"]]; then
```
Comment (jpsim) [inline build.sh]: `!=`?

Comment (segiddins) [inline build.sh]: silly me. fixed.

**Comment by jpsim** [comment]
Have you tried using this branch in CocoaPods locally? If so, :+1:

**Comment by segiddins** [comment]
Confirmed this works!

**Pull request merged**